OpenRA.Mods.Common/Traits/Buildings/Refinery.cs

```diff
@@ -162,25 +189,24 @@ void ITick.Tick(Actor self)
 		void INotifyActorDisposing.Disposing(Actor self)
 		{
 			CancelDock();
-			foreach (var harv in GetLinkedHarvesters())
-				harv.Trait.UnlinkProc(harv.Actor, self);
+			foreach (var harv in new List<Harvester>(linkedHarvesters))
```

Comment:
`ToArray`

Or better yet - let ``UnlinkHarvester` accept a boolean to indicate if the entry should be removed form the list. And after unlinking just call `Clear` on the list. Or rather create `UnlinkAllHarvesters`. Seems shame to copy the list - just to  avoid enumeration exception.